google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java

```diff
@@ -163,6 +170,76 @@ public Query limit(long limit) {
     return this;
   }
 
+  /**
+   * Split this query into multiple queries that can be evenly distributed across Bigtable nodes and
+   * be run im parallel. This method takes the results from {@link
+   * com.google.cloud.bigtable.data.v2.BigtableDataClient#sampleRowKeysAsync(String)} to divide this
+   * query into a set of disjoint queries that logically combine into form this query.
+   *
+   * <p>Expected Usage:
+   *
+   * <pre>{@code
+   * List<KeyOffset> keyOffsets = dataClient.sampleRowKeysAsync("my-table").get();
+   * List<Query> queryShards = myQuery.shard(keyOffsets);
+   * List<ApiFuture<List<Row>>> futures = new ArrayList();
+   * for (Query subQuery : queryShards) {
+   *   futures.add(dataClient.readRowsCallable().all().futureCall(subQuery))
+   * }
+   * List<List<Row>> results = ApiFutures.allAsList(futures).get();
```

Comment:
I think the carets need to be escaped or javadoc will barf

Reply:
Not sure what you mean. Do you mean that the `<>` need to be escaped? If so, I don't think thats true...`@code{}` renders them correctly

Reply:
All I know is that we escape them in our generated sample code because it caused a problem for us.

Reply:
The generated code samples use `<code>` instead of `{@code}`.  See here for a good breakdown:
https://reflectoring.io/howto-format-code-snippets-in-javadoc/#pre--code

The bigtable client  uses this style throughout, for example:
https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java#L142
and it seems to be working ok

Reply:
Ahh, `{@code}` doesn't handle `@` nicely, that must be why we used `<code>`. What a pain.